component/exporter.go

```diff
@@ -157,6 +157,13 @@ func WithLogsExporter(createLogsExporter CreateLogsExporterFunc) ExporterFactory
 	}
 }
 
+// WithExporterStabiityLevel overrides the default "unmaintained" stability level.
+func WithExporterStabilityLevel(sl StabilityLevel) ExporterFactoryOption {
```

Comment:
Stability level can potentially be defined independently for each signal type (so e.g. you can be beta for traces, stable for metrics and alpha for logs on a given exporter). How would one set this? (not sure if this is what you are alluding to in the README)

Reply:
@mx-psi right, this is what I meant with:

> the stability level is set per component factory, rather than per component. 

I'm open to suggestions on how to better solve this. I understand that in its current state, this would not work for the OTLP exporter for example, which is marked stable for traces/metrics but beta for logs.

Reply:
Okay, I wasn't sure if this is what you were talking about. 

Here is an alternative I can think of:

1. Add a new 'unsupported' stability level, which would be the zero value.
2. Replace `WithExporterStabilityLevel` with a new `WithExporterMetadata` passing the metadata:

```go
// ExporterMetadata defines human-readable metadata about the exporter
type ExporterMetadata struct {
  // StabilityMetrics defines the stability level for the metrics exporter.
  StabilityMetrics StabilityLevel
  // StabilityTraces defines the stability level for the traces exporter. 
  StabilityTraces StabilityLevel
  // StabilityLogs defines the stability level for the logs exporter.
  StabilityLogs StabilityLevel

  // Add further metadata in the future if we come up with anything worth putting here (e.g. warnings?)
}

func WithExporterMetadata(metadata ExporterMetadata) ExporterFactoryOption { 
  // ...
}
```

3. When applying a `With[Signal]Exporter` options, if `Stability[Signal]` is 'unsupported', replace it with 'unmaintained' instead.
4. After having applied all options, check that the metadata is consistent with the available create functions (i.e. that we don't mark as 'unsupported' anything that is actually supported).

This is complicated because we have to account for (i) in what order you pass the options and (ii) the possibility of mismatch between what one says in `ExporterMetadata` and what one passes in other options, but the public API is... satisfying? to me.

Reply:
I've added a `StabilityLevelUndefined` level, which is the default for any component that doesn't set the level for a particular signal. I've opted to go w/ a map as a parameter, this means in the future we'll be able to add additional signals without having to modify the struct.

PTAL at the updated implementation, it addresses the concern i had around not being able to set different stability levels for different components of the same factory.